src/gallery/gallery_screen.rs

```diff
@@ -155,27 +216,61 @@ impl Widget for Gallery {
 }
 
 impl Gallery {
-    fn calculate_current_offset(&self, padding: f64, width: f64, height: f64) -> DVec2 {
-        let padded_image_width = width + padding;
-        let padded_image_height = height + padding;
+    fn calculate_indexed_offset(&mut self, cx: &mut Cx) -> DVec2 {
+        let current_col = (self.current_index % self.grid_size) as f64;
+        let current_row = (self.current_index / self.grid_size) as f64;
+
+        let previous_col = (self.previous_index % self.grid_size) as f64;
+        let previous_row = (self.previous_index / self.grid_size) as f64;
+
+        let current_offset = dvec2(
+            -(IMAGE_WIDTH + IMAGE_PADDING) * current_col,
+            -(IMAGE_HEIGHT + IMAGE_PADDING) * current_row,
+        );
+        let previous_offset = dvec2(
+            -(IMAGE_WIDTH + IMAGE_PADDING) * previous_col,
+            -(IMAGE_HEIGHT + IMAGE_PADDING) * previous_row,
+        );
+
+        let padded_size = dvec2(
+            (IMAGE_WIDTH + IMAGE_PADDING),
+            (IMAGE_HEIGHT + IMAGE_PADDING),
```

Comment:
Getting this warning: `warning: unnecessary parentheses around function argument`